akka-actor/src/main/scala/akka/dispatch/ForkJoinExecutorConfigurator.scala

```diff
@@ -28,11 +29,22 @@ object ForkJoinExecutorConfigurator {
 
     override def execute(r: Runnable): Unit =
       if (r ne null)
-        super.execute(
-          (if (r.isInstanceOf[ForkJoinTask[_]]) r else new AkkaForkJoinTask(r)).asInstanceOf[ForkJoinTask[Any]])
+        super.execute(createTask(r))
       else
         throw new NullPointerException("Runnable was null")
 
+    def executeExternal(r: Runnable): Unit =
+      handle.invokeWithArguments(createTask(r))
+
+    private val handle = {
+      val m = classOf[ForkJoinPool].getDeclaredMethod("externalPush", classOf[ForkJoinTask[_]])
```

Comment:
Let's hope that this method is available on all supported JDKs...

Reply:
Looks like it was there already in 1.8 at least: https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/util/concurrent/ForkJoinPool.java#l1481